packages/storage/compat/service.ts

```diff
@@ -0,0 +1,91 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as types from '@firebase/storage-types';
+import { StorageService, isUrl, ref } from '../src/service';
+import { Location } from '../src/implementation/location';
+import { ReferenceCompat } from './reference';
+import { Reference } from '../src/reference';
+import { invalidArgument } from '../src/implementation/error';
+import { FirebaseApp } from '@firebase/app-types';
+
+/**
+ * A service that provides firebaseStorage.Reference instances.
+ * @param opt_url gs:// url to a custom Storage Bucket
+ */
+export class StorageServiceCompat implements types.FirebaseStorage {
+  constructor(
+    public app: FirebaseApp,
+    readonly _delegate: StorageService,
+    private _referenceConverter: (ref: Reference) => ReferenceCompat
+  ) {}
+
+  maxOperationRetryTime = this._delegate.maxOperationRetryTime;
+  maxUploadRetryTime = this._delegate.maxUploadRetryTime;
```

Comment:
Please use a getter here as well, as re-assigning these values will have no affect in the underlying SDK.

Reply:
Done.